testsuite/MDAnalysisTests/analysis/test_density.py

```diff
@@ -105,6 +113,76 @@ def test_check_set_unit_nolength(self, D):
         with pytest.raises(ValueError):
             D._check_set_unit(units)
 
+    def test_check_set_unit_density_none(self, D):
+        units = {'density': None}
+        assert D._check_set_unit(units) == None
```

Comment:
I might be missing the point of what this is doing here.
`_check_set_unit` has no return right? So asserting None is doesn't do anything?
Shouldn't you be asserting that `D.units['density'] is None`? (note that `is` is better than `==` for `None` checking).